### ISSUE TITLE
ADB Command problem

### ISSUE THREAD

**Issue by Akhenaten121** [theothernt/AerialViews#94]
When i tried to run the command from my mac "adb shell settings put secure screensaver_components com.neilturner.aerialviews/.ui.screensaver.DreamActivity" 

i got this  zsh: command not found: adb

how can i fix this?

**Comment by theothernt** [comment]
Just to confirm, have you installed the Android SDK platform tools?

**Comment by Akhenaten121** [comment]
Yes i did and connected to my device via USB cable nvidia ahield tv

**Comment by theothernt** [comment]
Ok, assuming you downloaded that .zip file to your `Downloads` folder, then extracted the files to `platform-tools`...

1. Open a new Terminal window
2. Type `cd Downloads/platform-tools` then press Return
3. Now type `./adb` and press Return

If that works, it means ADB is ok to use. Now you can type...

`./adb shell settings put secure screensaver_components com.neilturner.aerialviews/.ui.screensaver.DreamActivity`

**Comment by Akhenaten121** [comment]
T he extracted folder is on my desktop

**Comment by theothernt** [comment]
Ok, can you try this tool instead, it might be a bit easier to get working (I may update the instructions to use it instead)...

http://jocala.com/adblink2.html

**Comment by Akhenaten121** [comment]
That software not detecting my device

**Comment by Akhenaten121** [comment]
Now i'm getting this  ./adb: inaccessible or not found

**Comment by Akhenaten121** [comment]
Still getting adb: inaccessible or not found,  if i have USB connected do i still need to input the ip address?

**Comment by theothernt** [comment]
No, I don't think the USB connection will work without another piece of software installed.

On the Nvidia Shield, can you go into the Developer options menu and make sure that USB debugging is enabled?

Also, in the `platform-tools` folder on your Desktop, is there an adb file?

**Comment by Akhenaten121** [comment]
take a look at this:   ./adb shell settings put secure screensaver_components com.neilturner.aerialviews/.ui.screensaver.DreamActivity
* daemon not running; starting now at tcp:5037
* daemon started successfully

did it work?

**Comment by Akhenaten121** [comment]
> No, I don't think the USB connection will work without another piece of software installed.
> 
> On the Nvidia Shield, can you go into the Developer options menu and make sure that USB debugging is enabled?
> 
> Also, in the `platform-tools` folder on your Desktop, is there an adb file?

USB debugging is enabled and there is a terminal looking file name adb in the platform-tool folder

**Comment by theothernt** [comment]
> take a look at this: ./adb shell settings put secure screensaver_components com.neilturner.aerialviews/.ui.screensaver.DreamActivity
> 
> * daemon not running; starting now at tcp:5037
> * daemon started successfully
> 
> did it work?

That means the ADB command is working. To confirm the connection...

1. Type `./adb connect ip` where "ip" is the IP address of your Nvidia Shield
2. To confirm the connection type `./adb devices` which should list something like...

`List of devices attached`
`192.168.1.7:5555	device`

3. If that works, you can try `./adb shell settings put secure screensaver_components com.neilturner.aerialviews/.ui.screensaver.DreamActivity` again

**Comment by Akhenaten121** [comment]
> 3\. ./adb shell settings put secure screensaver_components com.neilturner.aerialviews/.ui.screensaver.DreamActivity

i got that   ./adb shell settings put secure screensaver_components com.neilturner.aerialviews/.ui.screensaver.DreamActivity
adb: more than one device/emulator

**Comment by theothernt** [comment]
Ok, try this...

1. Try `./adb disconnect` (disconnect from all devices)
2. Then `./adb connect 192.168.1.x` (connect to your Nvidia Shield only)

Then try the command above again?

**Comment by Akhenaten121** [comment]
> 3\. ./adb shell settings put secure screensaver_components com.neilturner.aerialviews/.ui.screensaver.DreamActivity

Still getting that more than 1 device emulator

**Comment by theothernt** [comment]
When to type `./adb devices`, what does it show?

**Comment by Akhenaten121** [comment]
> 3\. ./adb shell settings put secure screensaver_components com.neilturner.aerialviews/.ui.screensaver.DreamActivity

Ok how do i know if it works? i tried again and got  this.  ./adb shell settings put secure screensaver_components com.neilturner.aerialviews/.ui.screensaver.DreamActivity

**Comment by theothernt** [comment]
Actually, I should have asked... have you looked for the screensaver options on your Nvidia Shield?

https://www.nvidia.com/en-us/shield/support/shield-tv/sleep-timing-and-screen-saver/

**Comment by Akhenaten121** [comment]
> Actually, I should have asked... have you looked for the screensaver options on your Nvidia Shield?
> 
> https://www.nvidia.com/en-us/shield/support/shield-tv/sleep-timing-and-screen-saver/

Backdrop, Aerial Views and Colors  those are the options i have in screensaver

**Comment by theothernt** [comment]
Ok, you can just pick Aerial Views from that menu and that will set the screensaver as default.

Apologies for not mentioning this sooner, some devices don't have that menu which is why ADB commands are required - but not for the Nvidia Shield.

**Comment by theothernt** [comment]
Most of the common formats such as MOV, MKV and MP4.

By default, the app streams videos from the Apple's servers. You can turn this off and download the videos yourself from...

https://aerial-videos.netlify.app/

**Comment by Akhenaten121** [comment]
> Most of the common formats such as MOV, MKV and MP4.
> 
> By default, the app streams videos from the Apple's servers. You can turn this off and download the videos yourself from...
> 
> https://aerial-videos.netlify.app/

I eventually got it to work after turning of access to apple and other screensavers,  thank you very much for the help greatly appreciated

**Comment by theothernt** [comment]
Glad to hear that!

Feel free to create a new issue if you have questions in the future.
